### PR TITLE
Mantis 6863 trim headers

### PR DESCRIPTION
**Pull request by serge-gaia**
Trim headers when uploading csv or xlsx
https://decbugs.com/view.php?id=6863

### CODE REVIEW

File: biosys/apps/main/tests/api/test_serializers.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from django.test import TestCase
```
Comment (serge-gaia) [inline biosys/apps/main/tests/api/test_serializers.py:1]: Not related test. Testing that the API returns an error for dataset with not unique name.

File: biosys/apps/main/api/serializers.py [hunk 2]
```diff
@@ -57,8 +57,9 @@ def __call__(self, value):
             Dataset.validate_data_package(value, self.dataset_type)
 
         def set_context(self, serializer_field):
-            data = serializer_field.parent.context['request'].data
-            self.dataset_type = data.get('type')
+            if 'request' in serializer_field.parent.context:
```
Comment (serge-gaia) [inline biosys/apps/main/api/serializers.py:60]: Not related. Added a guard in the serializer

File: biosys/apps/main/api/uploaders.py [hunk 3]
```diff
@@ -81,6 +81,9 @@ def __init__(self, file_):
                 self.reader = csv.DictReader(codecs.iterdecode(self.file, 'utf-8'))
             else:
                 self.reader = csv.DictReader(self.file)
+        # because users are stupid we want to trim/strip the headers (fieldnames).
```
Comment (tony-gaia) [inline biosys/apps/main/api/uploaders.py:84]: I approve of this comment

**Comment by serge-gaia** [comment]
Test fixed

**Pull request merged**